File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/ServiceRequest.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class ServiceRequest : DomainResource, IResource
{
    public Base.CodeableConcept? PerformerType { get; set; }
    public Base.CodeableConcept[]? Category { get; set; }
    public Base.ResourceReference[]? Insurance { get; set; }
    public string[]? InstantiatesCanonical { get; set; }
    public string[]? InstantiatesUri { get; set; }
    public Base.ResourceReference[]? RelevantHistory { get; set; }
    public object? AsNeeded    
    {
        get
        {
            if (AsNeededBoolean is not null)
            {
                return AsNeededBoolean;
            }
    
            if (AsNeededCodeableConcept is not null)
            {
                return AsNeededCodeableConcept;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(bool))
            {
                AsNeededBoolean = (bool)value;return;
            }
    
            if (value?.GetType() == typeof(Base.CodeableConcept))
            {
                AsNeededCodeableConcept = (Base.CodeableConcept)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.ResourceReference[]? SupportingInfo { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public string? PatientInstruction { get; set; }
    public Base.ResourceReference[]? Specimen { get; set; }
    public Base.CodeableConcept[]? ReasonCode { get; set; }
    public string? AuthoredOn { get; set; }
    public string? OccurrenceTiming { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public bool? AsNeededBoolean { get; set; }
    public Base.Identifier? Requisition { get; set; }
    public Base.ResourceReference[]? LocationReference { get; set; }
    public Base.ResourceReference? Requester { get; set; }
    public string? Priority { get; set; }
    public Base.Period? OccurrencePeriod { get; set; }
    public required string Status { get; set; }
    public Base.Ratio? QuantityRatio { get; set; }
    public Base.CodeableConcept? Code { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public bool? DoNotPerform { get; set; }
    public Base.CodeableConcept[]? BodySite { get; set; }
    public required string Intent { get; set; }
    public Base.Range? QuantityRange { get; set; }
    public Base.Quantity? QuantityQuantity { get; set; }
    public Base.ResourceReference[]? Replaces { get; set; }
    public Base.CodeableConcept[]? OrderDetail { get; set; }
    public Base.ResourceReference[]? BasedOn { get; set; }
    public object? Quantity    
    {
        get
        {
            if (QuantityRatio is not null)
            {
                return QuantityRatio;
            }
    
            if (QuantityRange is not null)
            {
                return QuantityRange;
            }
    
            if (QuantityQuantity is not null)
            {
                return QuantityQuantity;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(Base.Ratio))
            {
                QuantityRatio = (Base.Ratio)value;return;
            }
    
            if (value?.GetType() == typeof(Base.Range))
            {
                QuantityRange = (Base.Range)value;return;
            }
    
            if (value?.GetType() == typeof(Base.Quantity))
            {
                QuantityQuantity = (Base.Quantity)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.CodeableConcept[]? LocationCode { get; set; }
    public string? OccurrenceDateTime { get; set; }
    public required Base.ResourceReference Subject { get; set; }
    public Base.CodeableConcept? AsNeededCodeableConcept { get; set; }
    public object? Occurrence    
    {
        get
        {
            if (OccurrenceTiming is not null)
            {
                return OccurrenceTiming;
            }
    
            if (OccurrencePeriod is not null)
            {
                return OccurrencePeriod;
            }
    
            if (OccurrenceDateTime is not null)
            {
                return OccurrenceDateTime;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                OccurrenceTiming = (string)value;return;
            }
    
            if (value?.GetType() == typeof(Base.Period))
            {
                OccurrencePeriod = (Base.Period)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OccurrenceDateTime = (string)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.ResourceReference[]? Performer { get; set; }
    public Base.ResourceReference[]? ReasonReference { get; set; }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/ServiceRequest.py
from typing import Optional, List
from base import Annotation
from base import Period
from base import CodeableConcept
from base import Range
from base import Timing
from base import Quantity
from base import DomainResource
from base import Ratio
from base import Reference
from base import Identifier

class ServiceRequest(DomainResource):
    performer_type: Optional[CodeableConcept] = None
    category: Optional[List[CodeableConcept]] = None
    insurance: Optional[List[Reference]] = None
    instantiates_canonical: Optional[List[str]] = None
    instantiates_uri: Optional[List[str]] = None
    relevant_history: Optional[List[Reference]] = None
    supporting_info: Optional[List[Reference]] = None
    encounter: Optional[Reference] = None
    patient_instruction: Optional[str] = None
    specimen: Optional[List[Reference]] = None
    reason_code: Optional[List[CodeableConcept]] = None
    authored_on: Optional[str] = None
    occurrence_timing: Optional[Timing] = None
    note: Optional[List[Annotation]] = None
    as_needed_boolean: Optional[bool] = None
    requisition: Optional[Identifier] = None
    location_reference: Optional[List[Reference]] = None
    requester: Optional[Reference] = None
    priority: Optional[str] = None
    occurrence_period: Optional[Period] = None
    status: str
    quantity_ratio: Optional[Ratio] = None
    code: Optional[CodeableConcept] = None
    identifier: Optional[List[Identifier]] = None
    do_not_perform: Optional[bool] = None
    body_site: Optional[List[CodeableConcept]] = None
    intent: str
    quantity_range: Optional[Range] = None
    quantity_quantity: Optional[Quantity] = None
    replaces: Optional[List[Reference]] = None
    order_detail: Optional[List[CodeableConcept]] = None
    based_on: Optional[List[Reference]] = None
    location_code: Optional[List[CodeableConcept]] = None
    occurrence_date_time: Optional[str] = None
    subject: Reference
    as_needed_codeable_concept: Optional[CodeableConcept] = None
    performer: Optional[List[Reference]] = None
    reason_reference: Optional[List[Reference]] = None